packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -1,53 +1,80 @@
-import { Models } from "@slicemachine/core";
-import * as Widgets from "./misc/widgets";
-import { snakelize } from "../utils/str";
-import { handleFields } from "./misc/handlers";
-import { FieldsSM } from "@slicemachine/core/build/models/Fields";
-import { SliceSM, VariationSM } from "@slicemachine/core/build/models/Slice";
+import {
+  SharedSlice,
+  SlicesTypes,
+  Variation,
+} from "@prismicio/types-internal/lib/customtypes/widgets/slices";
+import {
+  SharedSliceMockConfig,
+  VariationMockConfig,
+  generateSliceMock,
+  renderSliceMock,
+} from "@prismicio/mocks";
+import { buildFieldsMockConfig } from "./LegacyMockConfig";
+import { Slices, SliceSM } from "@slicemachine/core/build/models/Slice";
 
-const createEmptyMock = (
-  sliceName: string,
-  variation: VariationSM
-): Models.VariationMock => ({
-  variation: variation.id,
-  name: variation.name,
-  slice_type: snakelize(sliceName),
-  items: [],
-  primary: {},
-});
-
-export default function MockSlice(
-  sliceName: string,
-  model: SliceSM,
-  mockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
-): Models.SliceMock {
-  const variations = model.variations.map((variation) => {
-    const mock = createEmptyMock(sliceName, variation);
-    const handler: (
-      fields?: FieldsSM,
-      mocks?: Record<string, unknown>
-    ) => Models.VariationMock["primary"] = handleFields(Widgets);
-
-    const maybeMockConfig = mockConfig?.[variation.id]?.primary || {};
-    mock.primary = handler(variation.primary, maybeMockConfig);
-
-    const items: Models.VariationMock["items"] = [];
+function buildVariationMockConfig(
+  model: Variation,
+  legacyVariationMockConfig?: Partial<Record<string, Record<string, unknown>>>
+): VariationMockConfig {
+  const primaryFields =
+    model.primary &&
+    buildFieldsMockConfig(
+      model.primary,
+      legacyVariationMockConfig?.["primary"]
+    );
+  const itemFields =
+    model.items &&
+    buildFieldsMockConfig(model.items, legacyVariationMockConfig?.["items"]);
+  return {
+    nbItemsBlocks: 1,
```

Comment:
interesting, is this the number of times a block is repeated ?

Reply:
yes

Reply:
Nice, that could be awesome if later we want to allow the user to add that to the mock config.